roles/validations/tasks/edpm/hugepages_and_reboot.yml

```diff
@@ -0,0 +1,121 @@
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc get osdpns --no-headers -o custom-columns=":metadata.name"
+  register: deployed_nodeset_name
+
+  # collect initial confighash from the nodeset
+- name: Collecting initial confighash from the nodeset
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: >-
+      set -o pipefail && oc get -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} osdpns "{{ deployed_nodeset_name.stdout }}" -o yaml | yq .status.configHash
+  register: initial_config_hash
+
+- name: Update hugepages value
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc patch -n {{ cifmw_install_yamls_defaults['NAMESPACE'] }} osdpns/"{{ deployed_nodeset_name.stdout }}" --type=merge -p '{"spec": {"nodeTemplate": {"ansible": {"ansibleVars": {"edpm_kernel_args": "default_hugepagesz=1gb hugepagesz=1g hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"}}}}}'
+
+# The reboot-os service doesn't exist by default. So we need to create this service in
+# order to reboot the nodes.
+- name: Create reboot service
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: |
+      oc apply -f - <<EOF
+      apiVersion: dataplane.openstack.org/v1beta1
+      kind: OpenStackDataPlaneService
+      metadata:
+        name: reboot-os
+      spec:
+        playbook: osp.edpm.reboot
+      EOF
+
+# Create a new OpenStackDataPlaneDeployment to set the reboot variable. We also only need to execute
+# the jobs required to update Hugepages and reboot the nodes.
+- name: Create openstackdataplanedeployment to rollout changes
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.shell:
+    cmd: |
+      oc apply -f - <<EOF
+      apiVersion: dataplane.openstack.org/v1beta1
+      kind: OpenStackDataPlaneDeployment
+      metadata:
+        name: edpm-hugepages-update
+        namespace: {{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+      spec:
+        nodeSets:
+          - "{{ deployed_nodeset_name.stdout }}"
+        ansibleExtraVars:
+          edpm_reboot_force_reboot: "true"
+        servicesOverride:
+          - bootstrap
+          - reboot-os
+      EOF
+
+# loop check the status of the openstackdataplanedeployment until it is either completed,
+# or reaches a defined timeout value.
+- name: Wait for deployment to be complete
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: >-
+      oc wait openstackdataplanedeployment edpm-hugepages-update
+      --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }}
+      --for=condition=ready
+      --timeout={{ cifmw_validations_timeout | default(100) }}m
```

Comment:
enabling hugepages ideally woudl not require new task in roles

we should be able to do this in the ci framework by providing a customisation to the dataplen in the job

this can work but we probably should avoid special caseing thing like this with new tasks and try to be more genic since customer will not want to write and miantin playbooks to do multiple deploy like this.

Reply:
Hey Sean, I think you might be missing the context on this patch because the Github description isn't up to date with the commit message but this isn't for customer use.

Reply:
I'm not sure I'm picking up what you're putting down here. This is simply a QE validation job to test functionality of the dataplane-operator to take arbitrary kernel args as input, apply them to the edpm nodes and ensure the nodes are rebooted. I'm basically just paraphrasing my commit message though:
https://github.com/openstack-k8s-operators/ci-framework/pull/1356/commits/0876265aec15531002d18ab1e0459d355bcf0e41

In summary though, this change isn't about having the CI tests always test hugepages, it's just about testing functionality from a QE perspective.

So the idea being that any QE validation for any feature would exist in a single playbook, that can be executed using the framework defined in:
https://github.com/openstack-k8s-operators/ci-framework/pull/1326
https://issues.redhat.com/browse/OSPRH-5778

Reply:
what im saying is that oure QE jobs and the ci framework in general should try to emulate how a customer would do this.

the customer would do this as part of the initial greenfiled deployment ideally prior to the installation of ovs/nova

Reply:
i guess im pusshing back on the central premms of why this role exits

testing indigugal fution slike configuring hugepages should be tested in the edpm-ansible molecule test for the relevant role in this case this should be tested by the edpm_kernel molecule tests https://github.com/openstack-k8s-operators/edpm-ansible/tree/main/roles/edpm_kernel/molecule

Reply:
I guess the intention behind this QE test is the end-to-end capability of the product to take arbitrary user input via the `OpenStackDataPlaneNodeSet`, reconfigure the Ansible inventory, execute the required tasks to apply the arbitrary input, ensure the node is rebooted and the end result matches user expectations.

Testing in edpm_ansible alone doesn't guarantee the end-to-end functionality. It's good to have it there as well, but the whole solution needs to be validated. Consider the workflow required here, users first need to:

1. Update the `OpenStackDataPlaneNodeSet`;
2. The reboot service doesn't exist by default, so that needs to be created;
3. We need a new `OpenStackDataPlaneDeployment` to rollout the changes. Changes like this can be done in a limited subset of tasks, in this case just `bootstrap` and `reboot`. So we are also covering the `servicesOverride` functionality in the `OpenStackDataPlaneDeployment` controller.
4. New jobs need to be created, ansibleee needs to mount the updated inventory and execute the required playbooks.
5. `edpm_ansible` enters the picture to execute the `edpm_kernel` role to apply the new KernelArgs
6. System needs to reboot and the end result should be new kernel args in `/proc/cmdline`. 

`edpm_ansible` makes up a small component of the workflow in this case, and `edpm_kernel` is just a random role chosen to test the overall capability. We could choose any arbitrary ansible variable and role for testing this overall workflow.